src/Common/ThreadStatus.cpp

```diff
@@ -228,6 +226,9 @@ ThreadStatus::~ThreadStatus()
 
     chassert(!check_current_thread_on_destruction || current_thread == this);
 
+    /// Flush untracked_memory **before** switching the current_thread (to avoid losing untracked_memory)
```

Comment:
```suggestion
    /// Flush untracked_memory **right before** switching the current_thread to avoid losing untracked_memory in deleter (detachFromGroup)
```